working-with-components.md

```diff
@@ -0,0 +1,48 @@
+ - **Organisms:** These are groups of molecules joined together to form a relatively complex, distinct sections of our user interface. An example of an organism component may be a product tile, or a search header.
+
+At Holiday Extras we use Facebook's [React](https://facebook.github.io/react/index.html) for building composeable user interface components.
+
+## Why should we use components?
+Building a library of reuseable components has several benefits:
+
+ - Once built, the shared components can be reused in a fraction of the time they took to build.
+ - Shared components provide a consistent look and feel to all of Holiday Extras' user interfaces.
+ - Shared components allow user interface improvements to be rapidly rolled out across our websites and all of our customers and partners to benefit in a fraction of the time they otherwise would.
+ - Simple reuseable components can be rapidly composed together to build more complex user interfaces.
+ - Reuseable components vastly reduce code duplication in the front end.
+
+In the short term there's an obvious implemntation cost to setting all this up, but taking a long term view and keeping the above in mind, the question is really *"Why wouldn't we use components?"*.
+
+## Where and when do we use components?
+In short, we'll use them everywhere. Our goal is to be integrating reuseable components across all of our user interfaces. From our landing pages and search engines, to our web and mobile apps. In order to do this we'd like everyone to look at componentising user interfaces as they work on them.
+
+As for when to use components, the answer to this question can be broken down into two sections:
+
+### Writing new user interface code
+This should be componentised from the beginning, and this is the expectation for new pull requests. The UX/UI Team are on hand to offer advice and assistance with this if necessary.
+
+### Modifying existing user interface code
+When modifying existing user interface code, existing user interface elements that can be replaced by shared components we already have should be replaced. If there are no existing user interface components we should look at building some when time allows.
+
+
+## Where do we put shared components?
+Going back to the atomic design principles from the "What are components?" section we'd like to put our shared components in the following places:
+
+ - **Atom and Molecule Components:** These go in the public [UI-Toolkit](https://github.com/holidayextras/ui-toolkit) project where they can be reused everywhere. This is because at the granular atomic and molecular level, components probably have applicability across all of the Hx platforms as well as to the wider world. In support of this, the UI-Toolkit project is open source.
```

Comment:
" components probably have applicability across all of the Hx platforms as well as to the wider world. In support of this, the UI-Toolkit project is open source."

I don't like the use of the word *probably*, sorry but it does allows for ambiguity. Also we should probably not state HX, and just tates that:
`This is because at the granular atomic and molecular level, components have applicability across all uses in the web, they are not specific to HX and could be used by any company much the same way Bootstrap is not Twitter specific. In support of this, the UI-Toolkit project is open source.`